index.ts

```diff
@@ -200,16 +204,22 @@ cron.onEvent("partition-registrar", new CallbackFunction('partition-callback', {
     
         const client = athena.createClient(clientConfig, awsConfig);
     
-        const date = moment(event.time).utc().format("YYYY/MM/DD/HH");
-    
-        const query = `ALTER TABLE ${db.name.get()}.logs add
-        PARTITION (inserted_at = '${date}') LOCATION '${location.get()}/${date}/';`;
+        let date = moment(event.time);
+        
+        for(let i = 0; i <= 12; i++) {
+            const dateString = date.utc().format("YYYY/MM/DD/HH");
+
+            const query = `ALTER TABLE ${db.name.get()}.logs ADD IF NOT EXISTS
+            PARTITION (inserted_at = '${dateString}') LOCATION '${location.get()}/${dateString}/';`;
     
-        client.execute(query, (err: Error) => {
-            if (err) {
-                throw err;
-            }
-        })
+            client.execute(query, (err: Error) => {
+                if (err) {
+                    throw err;
+                }
```

Comment:
If you were to move this code into a a helper file, `/athena/partitionHelper.ts` and expose a method :

```ts
export const createPartitionDDLStatement(hours = 12): string {
...
}

```
then you could unit test this in `/athena/__tests__/partitionHelper.test.ts` and remove the need for modifying foo.test.

Reply:
Completed